scanner/cmd/scanner/main.go

```diff
@@ -178,6 +179,9 @@ func createGRPCService(backends *Backends, cfg *config.Config) (grpc.API, error)
 				ServeHTTP:      true,
 			},
 		},
+		// Setting this value causes the server to tell clients to GOAWAY after the specified duration (+/- some jitter).
+		// This is to ensure clients account for server-side horizontal scaling.
+		MaxConnectionAge: 5 * time.Minute,
```

Comment:
Do you know how this will behave in practice - curious under what scenarios will a `GOAWAY` occur that we need to account for (if at all). 

For example: if there is a request to indexer that has taken longer then 5 minutes, will the client open a new connection transparently in response to the `GOAWAY` for new requests? or will new requests all fail with `GOAWAY` until the long running request finishes?

Reply:
gRPC is built on top of HTTP/2 which tends to use a single, multiplexed connection for multiple requests. So, I expect the GOAWAY to be sent about every 5 minutes with this setting. Once this is sent, it tells the client's [gRPC name resolver](https://github.com/grpc/grpc-go/tree/master/examples/features/name_resolving#what-is-a-name-resolver) to update its list of known IPs. This allows the client to learn about new IPs in the case the number of server replicas increases.

Note: this will not cancel the current requests. The in-flight requests will still be allowed to finish. Just new ones will be affected

5 minutes may be too much because I don't know how long the HPA waits before it tries to scale down

Reply:
By the way, if the gRPC client allowed us to set our own DNS TTL (I'm not sure if it even respects whatever TTL K8S sends to it), we would not need this; we could simply make regular queries to get the most up-to-date list of servers and add them to the pool (scaling down should be handled by a graceful shutdown on the POD's side).

We just opted to cut off the conn because this is how we know right now how to force the resolver to perform another DNS query.

Reply:
This actually worth adding in the comments or PR description. In the current description and comments, it seems like cutting off the conn is expected to get a new list of IPs, which is not true.

Reply:
updated this whole PR because I found we were load balancing in an ideal way for our use-case. See the updated PR description and in-line comments in the code

as I was doing more research into why it looked like we were not originally load balancing properly (with the iteration of this PR originally reviewed), I cam across https://github.com/grpc/grpc-go/issues/3170#issuecomment-552517779 (also linked in the description) which says setting `MaxConnectionAge` is the way to go to ensure DNS records are updated on the client